code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -54,3 +54,108 @@
+			revenant_souls++
+			drained_mobs.Add(target)
+		else
+			user << "<span class='warning'>[target]'s soul is dead and empty.</span>"
+			return
+	..()
+
+obj/item/weapon/gun/magic/staff/staffofrevenant/attack_self(mob/living/user)
+	if(revenant_level == 0)
+		if(revenant_souls >= 5)
+			user << "<font color=purple>As you focus on the staff, you witness the crystal eminating a bright shine, before receeding again. The staff hums at an eerie tone, and has managed to become much stronger...</font>"
+			max_charges = 2
+			charges = 2
+			recharge_rate = 9
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level1(src)
+			revenant_level = 1
+			revenant_damage = 25
+	if(revenant_level == 1)
+		if(revenant_souls >= 10)
+			user << "<font color=purple>Once again, you glance at the staff, sparks now eminating from it as it begins to grow in power. You hear silent wailing around you, as you begin your descent into madness...</font>"
+			max_charges = 3
+			charges = 3
+			recharge_rate = 8
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level2(src)
+			revenant_level = 2
+			revenant_damage = 30
+			user << 'sound/spookoween/ghost_whisper.ogg'
+	if(revenant_level == 2)
+		if(revenant_souls >= 15)
+			user << "<font color=purple>You only give a quick glimpse at the staff, as you hear the screams of the fallen emenating from the staff's crystal. Your powers grow even stronger...</font>"
+			max_charges = 4
+			charges = 4
+			recharge_rate = 7
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level3(src)
+			revenant_level = 3
+			revenant_damage = 35
+			user << 'sound/hallucinations/veryfar_noise.ogg'
+	if(revenant_level == 3)
+		if(revenant_souls >= 20)
+			user << "<font color=purple>You only give a quick glimpse at the staff, as you hear the screams of the fallen emenating from the crystal mounted ontop of the staff, echoing throughout the station. Your powers grows even stronger...</font>"
+			max_charges = 5
+			charges = 5
+			recharge_rate = 6
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level4(src)
+			revenant_level = 4
+			revenant_damage = 40
+			world << 'sound/hallucinations/i_see_you1.ogg'
+			world << "<font color=purple><b>\"Your end is near...\"</font>"
+	if(revenant_level == 4)
+		if(revenant_souls >= 25) // if you reach this point, you pretty much won already
+			user << "<font color=purple>Just merely thinking of the power you have acquired is enough to trigger the staff's final evolution... It's destructive powers lets out an even louder wailing than last time, so loud that it echoes throughout the entire station, alerting those still standing that its futile to resist now...</font>"
+			max_charges = 6
+			charges = 6
+			recharge_rate = 5
+			chambered = new /obj/item/ammo_casing/magic/staffofrevenant/level5(src)
+			revenant_level = 5
+			revenant_damage = 60
+			world << "<font size=5><font color=purple><b>\"UNLIMITED... POWER!\"</font>"
```

Comment:
You can combine the attributes of size and color into one tag:

`<font size=5 color=purple>`